### PR TITLE
#176 AngularJS i18n support

### PR DESCRIPTION
**Pull request by Nazanin1369**

### CODE REVIEW

File: demo/resources/locale-zh_CN.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	"demo.styleGuide.p1": "您可以风格你的表单元素的不同状态使用下面的选择",
+	"demo.styleGuide.invalidSelectors": "无效的选择器",
+	"demo.styleGuide.youtStyleHere": "在这里你的风格",
+	"demo.styleGuide.validSelectors": "有效选择器",
+	"demo.styleGuide.p2": "您还可以包括你的供应商库类使用的是CSS预处理器，如",
+	"demo.styleGuide.or": "要么",
+	"demo.styleGuide.usingBootstrapClasses": "使用自举类",
+	"demo.styleGuide.p3": "欲了解更多有关AngularJS验证类检查官员AngularJS",
+	"demo.styleGuide.documentation": "文件"
+}
```
Comment (huei90) [inline demo/resources/locale-zh_CN.json]: Don't worry, I will do the translation job.

File: demo/demo.js [hunk 2]
```diff
@@ -1,20 +1,58 @@
...
+      'zh_CN': '汉语'
+    },
+    'preferredLocale': 'en_US'
+  })
 
   // -------------------
   // config phase
   // -------------------
-  .config(['$validationProvider', function($validationProvider) {
+  .config(['$validationProvider', '$translateProvider', 'tmhDynamicLocaleProvider', function($validationProvider, $translateProvider, tmhDynamicLocaleProvider, $translate) {
```
Comment (huei90) [inline demo/demo.js]: should this `$translate` need in last param?

File: demo/demo.js [hunk 3]
```diff
@@ -1,20 +1,58 @@
...
+
+    tmhDynamicLocaleProvider.localeLocationPattern('bower_components/angular-i18n/angular-locale_{{locale}}.js');
+
     var defaultMsg;
     var expression;
 
+    var successHTML = function(msg) {
+      return '<p class="validation-valid">' + this.$injector.get('$translate').instant(msg) + '</p>';
+    };
+
```
Comment (huei90) [inline demo/demo.js]: `this.xx` this looks good.

File: demo/demo.js (hunk 2 above)
Comment (Nazanin1369) [inline demo/demo.js]: Actually no. Forgot to remove it.

File: demo/demo.js (hunk 3 above)
Comment (Nazanin1369) [inline demo/demo.js]: Great!

**Comment by huei90** [comment]
I'm closing this, open if needed.

**Pull request closed without merging**